qiskit/visualization/pulse_v2/generators.py

```diff
@@ -0,0 +1,683 @@
+    color = _fill_waveform_color(channel)
+
+    # create real part
+    if any(re_y):
+        re_style = style.copy()
+        re_style['color'] = color.real
+        re_meta = meta.copy()
+        re_meta['data'] = 'real'
+        real = drawing_objects.FilledAreaData(data_type='Waveform',
+                                              channel=channel,
+                                              x=time,
+                                              y1=re_y,
+                                              y2=np.zeros_like(time),
+                                              meta=re_meta,
+                                              styles=re_style)
+        fill_objs.append(real)
+
+    # create imaginary part
+    if any(im_y):
+        im_style = style.copy()
+        im_style['color'] = color.imaginary
+        im_meta = meta.copy()
+        im_meta['data'] = 'imaginary'
+        imag = drawing_objects.FilledAreaData(data_type='Waveform',
+                                              channel=channel,
+                                              x=time,
+                                              y1=im_y,
+                                              y2=np.zeros_like(time),
+                                              meta=im_meta,
+                                              styles=im_style)
+        fill_objs.append(imag)
+
+    return fill_objs
+
+
+def gen_iqx_latex_waveform_name(inst_data: data_types.InstructionTuple) \
+        -> List[drawing_objects.TextData]:
+    r"""Generate formatted instruction name associated with the waveform.
+
+    Channel name and id are removed and the rotation angle is expressed in units of pi.
+    CR pulse name is also converted with the controlled rotation angle divided by 2.
+
+    Note that in many scientific articles the controlled rotation angle implies
+    the actual rotation angle, but in IQX backend the rotation angle represents
+    the difference between rotation angles with different control qubit states.
+
+    For example:
+        - 'X90p_d0_abcdefg' is converted into 'X(\frac{\pi}{2})'
+        - 'CR90p_u0_abcdefg` is converted into 'CR(\frac{\pi}{4})'
+
+    The `annotate` style is applied.
+
+    Notes:
+        This generator can convert pulse names used in the IQX backends.
+        If pulses are provided by the third party providers or the user defined,
+        the generator output may be the as-is pulse name.
+
+    Args:
+        inst_data: Waveform instruction data to draw.
+
+    Returns:
+        List of `TextData` drawing objects.
+    """
+    systematic_name = inst_data.inst.pulse.name
+
+    style = {'zorder': PULSE_STYLE.style['formatter.layer.annotate'],
+             'color': PULSE_STYLE.style['formatter.color.annotate'],
+             'size': PULSE_STYLE.style['formatter.text_size.annotate'],
+             'va': 'center',
+             'ha': 'center'}
+
+    template = r'(?P<op>[A-Z]+)(?P<angle>[0-9]+)?(?P<sign>[pm])_(?P<ch>[dum])[0-9]+'
```

Comment:
Where does this come from?

Reply:
this is from my observation. maybe I need to check the backend code if there is schema.